### PR TITLE
Bug/#9095 - Implement Audience Filter to Exclude Unavailable Audiences

### PR DESCRIPTION
**Pull request by hussain-t**
## Summary

Addresses issue:

- #9095 

## PR Author Checklist

- [x] My code is tested and passes existing unit tests.
- [x] My code has an appropriate set of unit tests which all pass.
- [ ] My code is backward-compatible with WordPress 5.2 and PHP 7.4.
- [ ] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [ ] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Code Reviewer Checklist

- [x] Run the code.
- [x] Ensure the acceptance criteria are satisfied.
- [x] Reassess the implementation with the IB.
- [x] Ensure no unrelated changes are included.
- [x] Ensure CI checks pass.
- [ ] Check Storybook where applicable.
- [x] Ensure there is a QA Brief.

## Merge Reviewer Checklist

- [x] Ensure the PR has the correct target branch.
- [x] Double-check that the PR is okay to be merged.
- [x] Ensure the corresponding issue has a ZenHub release assigned.
- [x] Add a changelog message to the issue.

### CODE REVIEW

File: assets/js/modules/analytics-4/components/audience-segmentation/dashboard/AudienceTilesWidget/index.test.js [hunk 1]
```diff
@@ -267,7 +267,36 @@ describe( 'AudienceTilesWidget', () => {
 		registry.dispatch( CORE_USER ).receiveGetAudienceSettings( {
 			configuredAudiences: [
 				'properties/12345/audiences/1',
-				'properties/12345/audiences/9',
+				'properties/12345/audiences/3',
```
Comment (techanvil) [inline assets/js/modules/analytics-4/components/audience-segmentation/dashboard/AudienceTilesWidget/index.test.js:270]: Hmm, this "should render when some configured audiences are matching available audiences" test is now identical to the "should render when all configured audiences are matching available audiences" test. It looks like we can simply remove this "should render when some..." test case.

**Review by techanvil — CHANGES_REQUESTED** [review]
Thanks @hussain-t, this is working well. Just one suggestion with regard to the tests, please take a look.

File: assets/js/modules/analytics-4/components/audience-segmentation/dashboard/AudienceTilesWidget/index.test.js (hunk 1 above)
Comment (hussain-t) [reply assets/js/modules/analytics-4/components/audience-segmentation/dashboard/AudienceTilesWidget/index.test.js]: Removed ✅

**Review by techanvil — APPROVED** [review]
LGTM, nice one @hussain-t!

**Pull request merged**